C/services/south/south.cpp

```diff
@@ -940,6 +959,86 @@ void SouthService::configChange(const string& categoryName, const string& catego
 	}
 }
 
+/**
+ * Separate thread to run plugin_reconf, to avoid blocking 
+ * service's management interface due to long plugin_poll calls
+ */
+static void reconfThreadMain(void *arg)
+{
+	SouthService *ss = (SouthService *)arg;
+	Logger::getLogger()->info("reconfThreadMain(): Spawned new thread for plugin reconf");
+	while(1)
+	{
+		ss->handlePendingReconf();
+	}
+	Logger::getLogger()->info("reconfThreadMain(): plugin reconf thread exiting");
+}
+
+/**
+ * Handle configuration change notification
+ */
+void SouthService::handlePendingReconf()
+{
+	Logger::getLogger()->debug("SouthService::handlePendingReconf: Going into cv wait");
+	mutex mtx;
+	unique_lock<mutex> lck(mtx);
+	m_cvNewReconf.wait(lck);
+	Logger::getLogger()->debug("SouthService::handlePendingReconf: cv wait has completed");
+
+	while(1)
```

Comment:
It's not clear what calls what and what is waiting and how the waits are interrupted for reconfiguration or shutdown.

Reply:
New reconf requests are queued up in SouthService::configChange() and m_cvNewReconf.notify_all() gets called to wake up the reconf thread's function SouthService::handlePendingReconf() and it processes all queued up reconf requests.